xwiki-platform-core/xwiki-platform-wysiwyg/xwiki-platform-wysiwyg-api/src/main/java/org/xwiki/wysiwyg/converter/RequestParameterConverter.java

```diff
@@ -40,13 +40,34 @@
 {
     /**
      * Check if the given request needs conversion and perform those conversions.
-     * This method is supposed to create a mutable request and to modify and returns that one. However in case of
-     * error it won't return the modified request, but it will handle directly the errors in the response.
+     * This method creates a mutable request, modifies and returns it. However in case of
+     * error it will return an empty optional, and it will handle directly the errors in the response.
+     * See {@link #convert(ServletRequest)} for using an exception for handling the errors.
      *
-     * @param request the request that might contain parameter needing conversion
+     * @param request the request that might contain parameter needing conversion or an {@link Optional#empty()} in case
+     *        of error
      * @param response the response used to redirect or do changes in case of conversion error
      * @return a mutable request with the converted parameters, or an empty optional in case of error
      * @throws IOException in case of problem to write an answer in the response
      */
     Optional<ServletRequest> convert(ServletRequest request, ServletResponse response) throws IOException;
+
+    /**
+     * Check if the given request needs conversion and perform those conversions.
+     * This method creates a mutable request, modifies it and returns it along with the errors and output that have
+     * occurred as part of the conversion, all that holds in the returned {@link RequestParameterConversionResult}.
+     * Consumer of this API should always check if the obtained result contains errors or not to know if the conversion
+     * properly succeeded.
+     *
+     * @param request the request that might contain parameter needing conversion
+     * @return an instance of {@link RequestParameterConversionResult} containing the modified request and the output
+     *         and errors that might have occurred
+     * @throws IOException in case of problem to write an answer in the response
+     * @since 14.10
+     */
+    @Unstable
+    default RequestParameterConversionResult convert(ServletRequest request) throws IOException
+    {
+        return null;
```

Comment:
imo it's better to have either no default implementation, or a exception thrown, instead of a null value.

Reply:
Yes indeed. Will change